### PR TITLE
feat: Upgrade to node 20

### LINKED ISSUE #196: Node 16 actions are deprecated
Warning from a recent run:
```
Node.js 16 actions are deprecated. Please update the following actions to use Node.js 20: inception-health/otel-export-trace-action@latest. For more information see: https://github.blog/changelog/2023-09-22-github-actions-transitioning-from-node-16-to-node-20/.
```

### PR DESCRIPTION
**Pull request by nicholascapo**
* Update to actions/checkout@v4
* Update to actions/setup-node@v4
* Set engines config in package.json
* Use node-version-file option for actions/setup-node
* Configure action to run using node20
* Fixes: #196

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -71,5 +71,8 @@
     "classNameTemplate": "{classname}",
     "includeConsoleOutput": "true",
     "reportTestSuiteErrors": "true"
+  },
+  "engines": {
+    "node": "^20"
```
Comment (nicholascapo) [inline package.json:76]: Need to confirm that this works correctly in CI.
I expect to get a node version that is `>=20, <21` but not exactly `20.0.0`

**Comment by nicholascapo** [comment]
Reviews please.
CC: @nikordaris @longility

**Review by nikordaris — APPROVED** [review]
@meleksomai LGTM

**Comment by nicholascapo** [comment]
Is there something I can do to help move this forward? :smile:

**Review by v1v — APPROVED** [review]

**Review by v1v — COMMENTED** [review]
For the record, this is a breaking change, see https://github.com/hashicorp/vault-action/issues/526.

If possible I'd say to release `v2`

**Comment by nicholascapo** [comment]
> For the record, this is a breaking change, see https://github.com/hashicorp/vault-action/issues/526.
> 
> If possible I'd say to release `v2`
> 
> 

That makes sense.
`v2` sounds good to me.

**Comment by nicholascapo** [comment]
@nikordaris are you able to approve this so that (at least) the CI runs? :smile:

**Comment by nicholascapo** [comment]
:wave: Any word here?
Anything I can do to help?

**Pull request closed without merging**